torchqdynamics/solver_utils.py

```diff
@@ -45,3 +45,15 @@ def bexpect(operators: Tensor, state: Tensor) -> Tensor:
     # TODO: Once QTensor is implemented, check if state is a density matrix or ket.
     # For now, we assume it is a density matrix.
     return torch.einsum('bij,...ji->...b', operators, state)
+
+
+def dissipator(rho: torch.Tensor, O: torch.Tensor) -> torch.Tensor:
+    return (
+        torch.einsum('abij,a...jk,abkl->a...il', O, rho, O.adjoint()) -
+        0.5 * torch.einsum('abij,abjk,a...kl->a...il', O.adjoint(), O, rho) -
+        0.5 * torch.einsum('a...ij,abjk,abkl->a...il', rho, O.adjoint(), O)
```

Comment:
A faster way to do this is to:
 - Compute `tmp = Ldag @ L @ rho`
 - Sum with its adjoint `tmp = tmp + tmp.adjoint()`
 - Add the jump term `rho = L @ rho @ Ldag - 0.5 * tmp`

All of this using einsum as much as possible of course. This runs only 4 matmuls, whereas your method runs 6. Also you can combine the 2nd and 3rd step in one line.